internal/processor/filtermetric/config.go

```diff
@@ -49,4 +49,8 @@ type MatchProperties struct {
 	// Expressions specifies the list of expr expressions to match metrics against.
 	// A match occurs if any datapoint in a metric matches at least one expression in this list.
 	Expressions []string `mapstructure:"expressions"`
+
+	// ResourceAttributes defines a map of resource attribute key to a list of possible values to match metrics against.
+	// A match occurs if any resource attribute matches at least one value in this given list.
+	ResourceAttributes map[string][]string `mapstructure:"resource_attributes"`
```

Comment:
For attributes matching we have `filterconfig.Attributes` see usage here https://github.com/open-telemetry/opentelemetry-collector/blob/master/internal/processor/filterconfig/config.go#L103, and the attribute matcher https://github.com/open-telemetry/opentelemetry-collector/blob/master/internal/processor/filtermatcher/attributematcher.go

Reply:
Makes sense. I will take a look.

Reply:
Just to clarify, I am suggesting to use the Attributes in the config instead of the map and delegate filtering to that matcher.

Reply:
Thanks Bogdan. I think I got your suggestion. Sorry for taking longer. I am kind of busy with our re:invent release works. Will address this soon (by this week).

Reply:
Hi @bogdandrutu , I want to discuss something. Following your suggestion I made it work. We can filter the metrics using resource attributes. However, it doesn't support our expected use case. Consider the following example- we want to get metrics from two containers (where the resource attribute key is same but the values are different). Right now, the `attributematcher` doesn't have a way to match one key against multiple values. It only does a strict match against the full resource attribute. 

To support our case, I think I need to change the whole structure of the `attributesmatcher`. Do you want me to introduce something new or modify the existing attributes matcher. 

Also, correct me if I am missing something. 

The following example doesn't work and I am not sure how can I achieve our goal with the existing solution.
```yaml
processors:
    filter:
        metrics:
            include:
                match_type: strict
                metric_names:
                    - container.memory.utilized
                    - container.memory.reserved
                resources:
                    - Key: container.name
                      Value: nginx200
                    - Key: container.name
                      Value: nginx300
```

Reply:
You can use a regexp matcher that will support that. Does that work for you?

Reply:
Ok, checking the `regexp` matcher.

Reply:
Hi @bogdandrutu , I played a little bit more with my changes using `attributesmatcher` and found that it already has support for `regexp` (which is `StringFilter` inside the matcher). I can utilize this feature to support our use case. Following configuration supports my expected use case. 

```yaml
processors:
    filter:
        metrics:
            include:
                match_type: regexp
                metric_names:
                    - container.memory.utilized
                    - container.memory.reserved
                resources:
                    - Key: container.name
                      Value: (nginx200|nginx300)

```
Thanks for your support. I will send an update after adding more unit tests.

Reply:
Hi @bogdandrutu , can you have a look. I pushed an update.